### ISSUE TITLE
Enriching projection by querying entity state (anti-pattern??)

### ISSUE THREAD

**Issue by aklikic** [akka/akka-platform-guide#377]
[VIII: Projection calling gRPC service - Call service from Projection](https://akka.io/platform-guide/microservices-tutorial/projection-grpc-client.html#_call_service_from_projection) (Scala and Java) is suggesting an enrichment pattern where entity is queried for the state information (`ShoppingCart.Get `command).
IMHO this pattern is breaking projection (event stream) consistency and could be even categorised as an anti-pattern for some other use cases. 
For a guide's use case (CheckOut), this pattern is not breaking consistency but we need to think about its implication on the reader. 
Implications of this pattern should at least be explicitly pointed out or this pattern should be removed and replaced with a pattern of enriching event(s) itself (when created in `ShoppingCart` entity).
Dedicated section on the projection enrichment patterns would also be nice to have.

**Comment by octonato** [comment]
I agree that in the case of shopping cart it's perfectly fine because CheckedOut is a final state. 

But I share your concern. I can pretty much imagine people getting it wrong. Fetching entity state inside a projection without realising that the state is most probably not in sync with the observed event. 
Maybe we should add a sentence saying, it's fine to fetch the entity state here because ...

As long as we don't go too deep, I think it's ok. We should avoid to open a discussion that could distract the reader.
